org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/uriresolver/CacheResourcesManager.java

```diff
@@ -258,6 +262,25 @@ private CompletableFuture<Path> downloadResource(final String resourceURI, Path
 		});
 	}
 
+	/**
+	 * Returns the http response code from a url connection, null if code could not
+	 * be retrived
+	 *
+	 * @param conn the URL connection
+	 * @return the http response code from a url connection, null if code could not
+	 *         be retrived
+	 */
+	private static String getHttpResponseCode(URLConnection conn) {
+		if (conn != null && conn instanceof HttpURLConnection) {
+			try {
+				return ((HttpURLConnection) conn).getResponseMessage();

```

Comment:
I think you want to call `getResponseCode()` here, which would return an integer.

See https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/net/HttpURLConnection.html#getResponseCode() .

Reply:
I could also do a combination of both where the message would be something like '404 Not Found'? I'm also looking for if there is a good way to test this case.

Reply:
Yeah, I like the combination of both actually. A little easier for users who may not recall.